src/algebra/big_operators/basic.lean

```diff
@@ -1322,6 +1334,37 @@ lemma prod_finset_coe [comm_monoid β] :
   ∏ (i : (s : set α)), f i = ∏ i in s, f i :=
 (finset.prod_subtype s (λ _, iff.rfl) f).symm
 
+@[to_additive]
+lemma prod_unique {α β : Type*} [comm_monoid β] [unique α] (f : α → β) :
+  (∏ x : α, f x) = f (default α) :=
+by rw [univ_unique, prod_singleton]
+
+@[to_additive]
+lemma prod_subsingleton {α β : Type*} [comm_monoid β] [subsingleton α] (f : α → β) (a : α) :
+  (∏ x : α, f x) = f a :=
+begin
+  haveI : unique α := { default := a, uniq := λ _, subsingleton.elim _ _ },
+  convert prod_unique f
+end
+
+lemma prod_unique_has_one {α β : Type*} [unique α] [has_one α] [comm_monoid β]
+  (f : α → β) :
+  ∏ x : α, f x = f 1 :=
+by rw [prod_unique f, unique_has_one]
```

Comment:
Does this work?
```suggestion
prod_subsingleton f 1
```
which to me suggests this lemma might not be worth having.

Reply:
Deleted.